server/methods/deleteMessage.coffee

```diff
@@ -18,7 +17,7 @@ Meteor.methods
 
 		deleteQuery = 
 			_id: message._id
-		deleteQuery['u._id'] = Meteor.userId() if user?.admin isnt true 
+		#deleteQuery['u._id'] = Meteor.userId() if user?.admin isnt true 
```

Comment:
Actually, by looking at the code above, we can see that we are already checking for permission:

`unless hasPermission or (deleteAllowed and deleteOwn)
  throw new Meteor.Error 'message-deleting-not-allowed', "[methods] deleteMessage -> Message deleting not allowed"`

`hasPermission` is checking if user is either admin or channel moderator: `RocketChat.authz.hasPermission(Meteor.userId(), 'delete-message', message.rid)`

Here's the permission:
`{ _id: 'delete-message', roles : ['admin', 'site-moderator', 'moderator']}`